### PR TITLE
Add e2e and unit tests for Key Manager Configuration Constraints feature

### PR DESCRIPTION
**Pull request by manodyaSenevirathne**
## Purpose 

$Subject

Fixes
- https://github.com/wso2/api-manager/issues/4734

## Description
Adds `00-WSO2-IS-km-constraints.spec.js` (E2E) and `constraintValidator.test.js`(Unit).

E2E: Covers 
- Creation of a WSO2-IS type key manager in Admin Portal with constraints
- Creation of an application in DevPortal
- Verify the creation of OAuth2 tokens using test values that adhere to and violate the specified constraints.
- Remove created application and the key manager

Unit: Covers MIN, MAX, RANGE, REGEX validation scenarios and edge cases.

## Related PRs
- https://github.com/wso2/apim-apps/pull/1246

## Related Issues
- https://github.com/wso2/api-manager/issues/4663

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for constraint validation covering MIN, MAX, RANGE, and REGEX validators, including edge cases (non-numeric, empty, negative, decimal), unknown/missing constraints, invalid regex handling, and hint message verification.
  * Added end-to-end tests for Key Manager constraint workflows: create/configure key manager, apply and enforce constraint settings, validate numeric input and error messaging during app/key generation, and automated cleanup.

### CODE REVIEW

File: tests/cypress/integration/e2e/keyManagerConstraints/00-WSO2-IS-km-constraints.spec.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2026, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may turn a copy of the License at
```
Comment (Piumal1999) [inline tests/cypress/integration/e2e/keyManagerConstraints/00-WSO2-IS-km-constraints.spec.js]: License header in this 2 newly added files are different. @manodyaSenevirathne Can you check and update the headers with the correct one?

File: tests/cypress/integration/e2e/keyManagerConstraints/00-WSO2-IS-km-constraints.spec.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        cy.contains('tr', kmName).within(() => {
+            cy.get('[aria-label="key-manager-delete-icon"]').click({ force: true });
+        });
+        cy.intercept('DELETE', '**/key-managers/**').as('deleteKM');
+        cy.get('[data-testid="form-dialog-base-save-btn"]', { timeout: 10000 })
+            .should('be.visible')
+            .click({ force: true });
+        cy.wait('@deleteKM', { timeout: 15000 }).its('response.statusCode').should('eq', 200);
+    });
+});
```
Comment (Piumal1999) [inline tests/cypress/integration/e2e/keyManagerConstraints/00-WSO2-IS-km-constraints.spec.js]: Add EOF

File: tests/cypress/integration/e2e/keyManagerConstraints/00-WSO2-IS-km-constraints.spec.js (hunk 1 above)
Comment (manodyaSenevirathne) [reply tests/cypress/integration/e2e/keyManagerConstraints/00-WSO2-IS-km-constraints.spec.js]: fixed in [00d8e3f](https://github.com/wso2/apim-apps/pull/1260/commits/00d8e3ff1489e9d454768cc83e21afec474c82a6)

**Review by Piumal1999 — APPROVED** [review]

**Comment by manodyaSenevirathne** [comment]
> @manodyaSenevirathne, the `00-WSO2-IS-km-constraints.spec.js` test is failing in the github action.

This was an intermittent failure , succeeded later
